x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/rest/action/profile/RestDisableProfileAction.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+package org.elasticsearch.xpack.security.rest.action.profile;
+
+import org.elasticsearch.action.support.WriteRequest;
+import org.elasticsearch.client.internal.node.NodeClient;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.license.XPackLicenseState;
+import org.elasticsearch.rest.RestRequest;
+import org.elasticsearch.rest.action.RestToXContentListener;
+import org.elasticsearch.xpack.core.security.action.profile.SetProfileEnabledAction;
+import org.elasticsearch.xpack.core.security.action.profile.SetProfileEnabledRequest;
+import org.elasticsearch.xpack.security.rest.action.SecurityBaseRestHandler;
+
+import java.io.IOException;
+import java.util.List;
+
+import static org.elasticsearch.rest.RestRequest.Method.POST;
+import static org.elasticsearch.rest.RestRequest.Method.PUT;
+
+public class RestDisableProfileAction extends SecurityBaseRestHandler {
+
+    public RestDisableProfileAction(Settings settings, XPackLicenseState licenseState) {
+        super(settings, licenseState);
+    }
+
+    @Override
+    public List<Route> routes() {
+        return List.of(
+            Route.builder(POST, "/_security/profile/_disable/{uid}").build(),
+            Route.builder(PUT, "/_security/profile/_disable/{uid}").build()
```

Comment:
I did a quick code search and it looks like the resource name/id comes after the resource type, ie profile .
I hadn't notice the update profile API until now.
My preference would be to have the `uid` first, but I'll leave the decision up to you, it's style only.

Reply:
TBC my preference is based on consistency grounds. eg `_security/role/{name}`, `_security/user/{id}`

Reply:
I agree on being consistency and have changed the URL format to be `uid` first. I'll have a separate PR for changing the UpdateProfileData API since it needs to touch a few more places like docs, rest spec etc. I also have a few other tweaks in mind which can be grouped together in the separate PR.